lib/init.js

```diff
@@ -116,6 +116,8 @@ async function init (config = {}) {
     case 'cosmos':
       if (config.cosmos) {
         logger.debug('init with user provided cosmosDB credentials')
+        // Do not reuse cosmos client instances for BringYourOwn creds
+        CosmosStateStore.inMemoryInstance = {}
```

Comment:
I think this can be in `CosmosStateStore.init`. This way any code trying to call its init will not have to worry about whether it should reuse client or not.

Reply:
Not really. If this is put in `CosmosStateStore.init`, there won't be any reuse at all. The idea here is to `not` reuse only for BringYourOwn cases.